src/terrain/terrain.js

```diff
@@ -811,18 +812,26 @@ export class Terrain extends Elevation {
         const fb = context.createFramebuffer(bufferSize[0], bufferSize[1], false);
         fb.colorAttachment.set(tex.texture);
 
+        const renderbuffer = depthAttachment ? depthAttachment : context.createRenderbuffer(context.gl.DEPTH_STENCIL, fb.width, fb.height);
+        fb.depthAttachment = new DepthStencilAttachment(context, fb.framebuffer);
+        fb.depthAttachment.set(renderbuffer);
+        if (!depthAttachment) {
+            context.clear({stencil: 0});
+            this._stencilRef = 0;
+        }
+
         if (context.extTextureFilterAnisotropic && !context.extTextureFilterAnisotropicForceOff) {
             gl.texParameterf(gl.TEXTURE_2D,
                 context.extTextureFilterAnisotropic.TEXTURE_MAX_ANISOTROPY_EXT,
                 context.extTextureFilterAnisotropicMax);
         }
 
-        return {fb, tex, dirty: false, ref: 1};
+        return {fb, tex, dirty: false};
     }
 
     _initFBOPool() {
         while (this.pool.length < Math.min(FBO_POOL_SIZE, this.proxyCoords.length)) {
-            this.pool.push(this._createFBO());
+            this.pool.push(this._createFBO(this.pool.length > 0 ? this.pool[0].fb.depthAttachment.current : null));
```

Comment:
Pushed a small simplification to make it more explicit that we're sharing the depth stencil between these framebuffers with this change in https://github.com/mapbox/mapbox-gl-js/commit/e9768c5b05dade82c4fcb485f0fac3b70389804d (with `this.pool.length > 0 ? this.pool[0].fb.depthAttachment.current : null)` here and `this.pool[index % FBO_POOL_SIZE].fb.depthAttachment.current` line 1001, I didn't find it immediately obvious where the depth attachment was coming from at framebuffer creation time).